google-cloud-storage/src/main/scala/akka/stream/alpakka/googlecloud/storage/GCStorageSettings.scala

```diff
@@ -0,0 +1,180 @@
+  def getBasePath: String = basePath
+
+  /** Java API */
+  def getTokenUrl: String = tokenUrl
+
+  /** Java API */
+  def getTokenScope: String = tokenScope
+
+  def withProjectId(value: String): GCStorageSettings = copy(projectId = value)
+  def withClientEmail(value: String): GCStorageSettings = copy(clientEmail = value)
+  def withPrivateKey(value: String): GCStorageSettings = copy(privateKey = value)
+  def withBaseUrl(value: String): GCStorageSettings = copy(baseUrl = value)
+  def withBasePath(value: String): GCStorageSettings = copy(basePath = value)
+  def withTokenUrl(value: String): GCStorageSettings = copy(tokenUrl = value)
+  def withTokenScope(value: String): GCStorageSettings = copy(tokenScope = value)
+
+  private def copy(
+      projectId: String = projectId,
+      clientEmail: String = clientEmail,
+      privateKey: String = privateKey,
+      baseUrl: String = baseUrl,
+      basePath: String = basePath,
+      tokenUrl: String = tokenUrl,
+      tokenScope: String = tokenScope
+  ): GCStorageSettings = new GCStorageSettings(
+    projectId = projectId,
+    clientEmail = clientEmail,
+    privateKey = privateKey,
+    baseUrl = baseUrl,
+    basePath = basePath,
+    tokenUrl = tokenUrl,
+    tokenScope = tokenScope
+  )
+
+  override def toString =
+    "GCStorageSettings(" +
+    s"projectId=$projectId," +
+    s"clientEmail=$clientEmail," +
+    s"privateKey=$privateKey," +
+    s"baseUrl=$baseUrl," +
+    s"basePath=$basePath," +
+    s"tokenUrl=$tokenUrl," +
+    s"tokenScope=$tokenScope" +
+    ")"
+
+  override def equals(other: Any): Boolean = other match {
+    case that: GCStorageSettings =>
+      java.util.Objects.equals(this.projectId, that.projectId) &&
+      java.util.Objects.equals(this.clientEmail, that.clientEmail) &&
+      java.util.Objects.equals(this.privateKey, that.privateKey) &&
+      java.util.Objects.equals(this.baseUrl, that.baseUrl) &&
+      java.util.Objects.equals(this.basePath, that.basePath) &&
+      java.util.Objects.equals(this.tokenUrl, that.tokenUrl) &&
+      java.util.Objects.equals(this.tokenScope, that.tokenScope)
+    case _ => false
+  }
+
+  override def hashCode(): Int =
+    java.util.Objects.hash(projectId, clientEmail, privateKey, baseUrl, basePath, tokenUrl, tokenScope)
+}
+
+object GCStorageSettings {
+  val ConfigPath = "alpakka.googlecloud.storage"
+
+  private val defaultBaseUrl = "https://www.googleapis.com/"
+  private val defaultBasePath = "/storage/v1"
+  private val defaultTokenUrl = "https://www.googleapis.com/oauth2/v4/token"
+  private val defaultTokenScope = "https://www.googleapis.com/auth/devstorage.read_write"
+
+  /**
+   * Reads from the given config.
+   */
+  def apply(c: Config): GCStorageSettings = {
+    val projectId = c.getString("project-id")
+    val clientEmail = c.getString("client-email")
+    val privateKey = c.getString("private-key")
+    val baseUrl = if (c.hasPath("base-url")) c.getString("base-url") else defaultBaseUrl
```

Comment:
The idea of using config is to have all defaults in `reference.conf` so that they are always set.